vmaas/reposcan/database/csaf_store.py

```diff
@@ -46,7 +49,76 @@ def _save_csaf_files(self, csaf_files: CsafFiles) -> list[int]:
         self.conn.commit()
         return db_ids
 
+    def _insert_missing_products(self, product_list):
+        product_tuples = [product.to_tuple() for product in product_list]
+        try:
+            cur = self.conn.cursor()
+            if product_tuples:
+                execute_values(cur, """
+                                    INSERT INTO csaf_products (cpe, package, module)
+                                    SELECT v.cpe, v.package, v.module
+                                    FROM (VALUES %s) AS v(cpe, package, module)
+                                    ON CONFLICT (cpe, package, module) DO NOTHING;
+                                    """, (product_tuples))
+                self.conn.commit()
+        except Exception as exc:
+            CSAF_FAILED_IMPORT.inc()
+            self.logger.exception("Failure while inserting data into csaf_products: %s", exc)
+            self.conn.rollback()
+        finally:
+            cur.close()
+
+    def _insert_cves(self, cve, product_list):
+        product_tuples = [product.to_tuple() for product in product_list]
+        try:
+            cur = self.conn.cursor()
+            if product_tuples:
+                execute_values(cur, f"""
+                    WITH product_ids AS (
+                        SELECT id
+                        FROM csaf_products
+                        WHERE (cpe, package, module) IN %s
+                    )
+                    INSERT INTO csaf_cves (cve, csaf_product_id, product_status_id)
+                    SELECT '{cve}', id, {PRODUCT_STATUS_ID}
+                    FROM product_ids
+                    ON CONFLICT (cve, csaf_product_id)
+                    DO UPDATE SET product_status_id = EXCLUDED.product_status_id;
+                    """, (product_tuples,))
+            self.conn.commit()
+        except Exception as exc:
+            CSAF_FAILED_IMPORT.inc()
+            CSAF_FAILED_UPDATE.inc()
+            self.logger.exception("Failure while inserting or updating data to csaf_cves: %s", exc)
+            self.conn.rollback()
+        finally:
+            cur.close()
+
+    def _populate_cves(self, csaf_cves):
+        for cve, product_list in csaf_cves.items():
+            self._insert_missing_products(product_list)
+            self._insert_cves(cve, product_list)
```

Comment:
Yes, I agree, deleting obsolete rows from `csaf_cves` for a given CVE has to be done here (or inside `_insert_cves`). Otherwise `_delete_unreferenced_products` has no purpose.

Don't confuse it with deleting whole CVEs which can't be done yet.